src/shared/acl/acl.service.ts

```diff
@@ -2,21 +2,20 @@ import { ROLE } from './../../auth/constants/role.constant';
 import { AclRule, RuleCallback } from './acl-rule.constant';
 import { Action } from './action.constant';
 import { Actor } from './actor.constant';
-import { Resource } from './resource.constant';
 
-export class BaseAclService {
+export class BaseAclService<T> {
```

Comment:
Should we use a more verbose name for this type? Like `Resource`

Reply:
@Seb-C 
Calling the generic type as `Resource` sounds good too.

But I was thinking calling it `T` makes it more explicit and less chances of people failing to notice what's going on here.
But I am cool with either. 

What do you think?

Reply:
I think it's not obvious what `T` is supposed to be (unless someone already knows the codebase).

Also, I'm wondering that it will become more confusing if we ever need multiple generic types in this class.

Reply:
> unless someone already knows the codebase
Okay, very valid point.

> if we ever need multiple generic types 
Yeah, we need to think of this scenario too.